.buildbot.sh

```diff
@@ -33,5 +21,27 @@ sh rustup.sh --default-host x86_64-unknown-linux-gnu \
     --profile minimal \
     -y
 export PATH=${CARGO_HOME}/bin/:$PATH
+
+# Build the compiler and add it as a linked toolchain.
+git clone ${YKRUSTC_REPO}
+cd ykrustc
+cp .buildbot.config.toml config.toml
+./x.py build --stage 1
+rustup toolchain link ykrustc-stage1 `pwd`/build/x86_64-unknown-linux-gnu/stage1
+cd ..
+rustup override set ykrustc-stage1
+
+# Test both workspaces using the compiler we just built.
+export RUSTFLAGS="-C tracer=${CI_TRACER_KIND} -D warnings"
+cargo xtask test
+cargo xtask bench
+cargo xtask clean
+
+# Also test the build without xtask, as that's what consumers will do.
+cargo build
+
+# Run rustfmt.
```

Comment:
Given how lengthy the preceding stages are, should we do `fmt` first? We also won't have to unset `RUSTFLAGS` that way!

Reply:
I wondered that myself. But can we do it as a separate PR, as testing these things properly it time consuming.

Reply:
TBH, I'm happy with you raising the commit and doing `bors try`1